tests/helpers.py

```diff
@@ -20,4 +41,269 @@ def create_unique_resource_name(resource_description, resource_type):
     current_date_time = (
         datetime.datetime.now().strftime("%d%H%M%S%f")
     )
-    return f"{resource_type}_{resource_description[:23]}_{current_date_time[:10]}"
+    return (
+        f"{resource_type}-{resource_description[:23]}-{current_date_time[:10]}"
+    )
+
+
+def create_resource(
+    resource_kind, resource_name, project_name, template_yaml,
+    desired_status='Available', wait=True, **kwargs
+):
+    """
+    Create a resource
+
+    Args:
+        resource_kind (str): The kind of the resource (e.g. CephBlockPool)
+        resource_name (str): The name of the resource
+        project_name (str): The name of the project that the new resource
+            should be associated with
+        template_yaml (str): The yaml file to use (e.g. CephBlockPool.yaml)
+        desired_status (str): The status of the resource to wait for
+        wait (bool): True for waiting for the resource to be created
+        **kwargs: Keyword args to be set for the new created resource
+
+    Raises:
+        AssertionError: In case of any failure
+    """
+    template = os.path.join(TEMPLATES_DIR, template_yaml)
+    templating.dump_to_temp_yaml(template, TEMP_YAML, **kwargs)
+    ocp_obj = ocp.OCP(kind=resource_kind, namespace=project_name)
+    logger.info(f"Creating a {resource_kind} {resource_name}")
+    assert ocp_obj.create(yaml_file=TEMP_YAML), (
+        f"Failed to create {resource_kind} {resource_name}"
+    )
+    assert ocp_obj.get(resource_name=resource_name), (
+        f"Failed to create {resource_kind} {resource_name}"
+    )
+    if wait:
+        assert ocp_obj.wait_for_resource(
+            condition=desired_status, resource_name=resource_name
+        ), f"{resource_kind} {resource_name} failed to reach status Available"
+        delete_file(TEMP_YAML)
+
+
+def delete_resource(
+    resource_kind, resource_name, project_name, wait=True
+):
+    """
+    Delete a resource
+
+    Args:
+        resource_kind (str): The kind of the resource (e.g. CephBlockPool)
+        resource_name (str): The name of the resource
+        project_name (str): The name of the project that the resource
+            is associated with
+        wait (bool): True for wait, False otherwise
+
+    Raises:
+        AssertionError: In case of any failure
+    """
+    ocp_obj = ocp.OCP(kind=resource_kind, namespace=project_name)
+    if ocp_obj.get(resource_name=resource_name):
+        logger.info(f"Deleting {resource_kind} {resource_name}")
+        assert ocp_obj.delete(resource_name=resource_name, wait=False), (
+            f"Failed to delete {resource_kind} {resource_name}"
+        )
+        if wait:
+            assert ocp_obj.wait_for_resource(
+                condition='', resource_name=resource_name, to_delete=True
+            ), f"{resource_kind} {resource_name} still exists'"
+            delete_file(TEMP_YAML)
```

Comment:
I doubt if this should be placed within 'if wait' block? This means TEMP_YAML will be deleted only when wait = true, isn't it? is that the intention?

Reply:
@keesturam Good catch!